packages/boot/test/bootstrapTests/upgradeAPI.test.ts

```diff
@@ -0,0 +1,311 @@
+      },
+    },
+    governedApis: ['governanceApi'],
+  };
+
+  const governedInstallation = await installBundle(governedContractBundle);
+  await installBundleToVatAdmin(governedContract2Bundle);
+  const governorTerms = {
+    timer,
+    governedContractInstallation: governedInstallation,
+    governed: {
+      terms: governedTerms,
+      issuerKeywordRecord: {},
+    },
+  };
+
+  const governorFacets = await EV(zoe).startInstance(
+    governorInstallation,
+    {},
+    governorTerms,
+    {
+      governed: {
+        initialPoserInvitation: poserInvitation,
+      },
+    },
+  );
+
+  return {
+    governorFacets,
+    invitationAmount,
+    voteCounterInstallation,
+    contract2SHA: governedContract2Bundle.endoZipBase64Sha512,
+    poserInvitation2,
+  };
+};
+
+// A more minimal set would be better. We need governance, but not econ vats.
+const PLATFORM_CONFIG = '@agoric/vm-config/decentral-main-vaults-config.json';
+
+const makeDefaultTestContext = async t => {
+  console.time('DefaultTestContext');
+  const swingsetTestKit = await makeSwingsetTestKit(t.log, undefined, {
+    configSpecifier: PLATFORM_CONFIG,
+  });
+
+  const { runUtils, storage, controller } = swingsetTestKit;
+  console.timeLog('DefaultTestContext', 'swingsetTestKit');
+  const { EV } = runUtils;
+  const zoe: ZoeService = await EV.vat('bootstrap').consumeItem('zoe');
+  const timer = await EV.vat('bootstrap').consumeItem('chainTimerService');
+
+  // has to be late enough for agoricNames data to have been published
+  const agoricNamesRemotes = makeAgoricNamesRemotesFromFakeStorage(storage);
+
+  const walletFactoryDriver = await makeWalletFactoryDriver(
+    runUtils,
+    storage,
+    agoricNamesRemotes,
+  );
+
+  const governanceDriver = await makeGovernanceDriver(
+    swingsetTestKit,
+    agoricNamesRemotes,
+    walletFactoryDriver,
+    wallets,
+  );
+
+  const facets = await setUpGovernedContract(zoe, timer, EV, controller);
+
+  return { ...swingsetTestKit, facets, governanceDriver };
+};
+
+const test = anyTest as TestFn<
+  Awaited<ReturnType<typeof makeDefaultTestContext>>
+>;
+
+test.before(async t => {
+  t.context = await makeDefaultTestContext(t);
+});
+
+test.after.always(t => {
+  return t.context.shutdown && t.context.shutdown();
+});
+
+test.serial(`start contract; verify`, async t => {
+  const { runUtils, facets } = t.context;
+  const {
+    governorFacets: { creatorFacet },
+  } = facets;
+  const { EV } = runUtils;
+  const contractPublicFacet = await EV(creatorFacet).getPublicFacet();
+
+  const avogadro = await EV(contractPublicFacet).getNum();
+  t.is(await EV(contractPublicFacet).getApiCalled(), 0);
+  t.is(avogadro, 602214090000000000000000n);
+});
+
+const getQuestionId = id => `propose-question-${id}`;
+const getVoteId = id => `vote-${id}`;
+
+const offerIds = {
+  invite: { charter: 'ch', committee: 'ctt' },
+  add1: { outgoing: 'add1' },
+  add2: { outgoing: 'add2' },
+};
+
+let governedContract;
```

Comment:
module level mutable state. hm. consider using a promise(kit) in `t.context` something like `t.context.governedContractKit.promise` and `.resolver`